Sources/StreamChat/Controllers/DatabaseObserver/ListDatabaseObserver.swift

```diff
@@ -252,12 +252,12 @@ class ListDatabaseObserver<Item, DTO: NSManagedObject> {
                   let context = self?.context else { return [] }
             var result: LazyCachedMapCollection<Item>!
             context.performAndWait {
-                result = LazyCachedMapCollection(
-                    source: frc.fetchedObjects ?? [],
-                    itemCreator: itemCreator,
-                    sorting: sorting,
-                    context: context
-                )
+                let dtos = frc.fetchedObjects ?? []
+                var mapped = dtos.compactMapLoggingError { try itemCreator($0) }
```

Comment:
Old code had the assert which caused the crash (see the LazyCachedMapCollection init for more information), new code logs the error and skips the element.